linker/shared/src/main/scala/org/scalajs/linker/frontend/optimizer/OptimizerCore.scala

```diff
@@ -731,12 +731,12 @@ private[optimizer] abstract class OptimizerCore(
         val replacement = ReplaceWithVarRef(newName, newSimpleState(Unused), None)
         val localDef = LocalDef(tcaptureValue.tpe, mutable, replacement)
         val localIdent = LocalIdent(newName)(ident.pos)
-        val newParamDef = ParamDef(localIdent, newOriginalName, ptpe, mutable)(paramDef.pos)
+        val newParamDef = ParamDef(localIdent, newOriginalName, tcaptureValue.tpe.base, mutable)(paramDef.pos)
 
         /* Note that the binding will never create a fresh name for a
          * ReplaceWithVarRef. So this will not put our name alignment at risk.
          */
-        val valueBinding = Binding.temp(paramName, ptpe, mutable, tcaptureValue)
+        val valueBinding = Binding.temp(paramName, tcaptureValue.tpe.base, mutable, tcaptureValue)
```

Comment:
Given that `mutable == false` (per assertion), this is now equivalent to
```suggestion
        val valueBinding = Binding.temp(paramName, tcaptureValue)
```